[DOC] Logging config updates

PaulRMellor opened the pull request:
Signed-off-by: prmellor <pmellor@redhat.com>

- Documentation

### Description
Updates to logging sections to make the logging config requirements clearer and easier to find.

**Kafka**
- Created new con-kafka-logging.adoc 
- Deleted ref-loggers-kafka.adoc
- Deleted snip-inline-logging-kafka.adoc 
- Deleted snip-configmap-logging-kafka.adoc
- Updated assembly-deployment-configuration-kafka.adoc

**Kafka Bridge**
- Created new con-kafka-bridge-logging.adoc
- Deleted ref-loggers-KafkaBridge.adoc
- Deleted snip-inline-logging-KafkaBridge.adoc
- Deleted snip-configmap-logging-kafkaBridge.adoc
- Updated assembly-deployment-configuration-kafkaBridge.adoc

**Kafka Connect**
- Created new con-kafka-connect-logging.adoc
- Deleted ref-loggers-Kafkaconnect.adoc
- Deleted snip-inline-logging-KafkaConnect.adoc
- Deleted snip-configmap-logging-kafkaConnect.adoc
- Updated assembly-deployment-configuration-kafkaconnect.adoc

**Kafka Connect S2I**
- Created new con-kafka-connect-s2i-logging.adoc
- Deleted ref-loggers-KafkaconnectS2i.adoc
- Deleted snip-inline-logging-KafkaConnectS2I.adoc
- Deleted snip-configmap-logging-kafkaConnectS2I.adoc
- Updated assembly-deployment-configuration-kafkaconnectS2I.adoc

**Common**
- Deleted proc-kafka-inline-logging.adoc
- Deleted proc-kafka-external-logging.adoc
- Deleted assembly-logging.adoc
- Deleted Appendix A: Configurable loggers -- ref-kafka-logging.adoc
- Updated master.adoc to remove the appendix

**Operators**
- Created section for Entity Operator operator loggers -- con-entity-operator-logging.adoc
- Updated EO configuration section -- ref-kafka-entity-operator.adoc
- Restructured Entity Operator config section, so that the User Operator and Topic Operator sections move up a level with new files: con-configuring-user-operator.adoc and con-configuring-topic-operator.adoc

log4j and log4j2 implmentations specified for each component.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

PaulRMellor commented:
NOTE: There is a separate task planned to provide more information on using configmaps for external logging.

scholzj commented on documentation/modules/con-configuring-topic-operator.adoc:
```diff
@@ -0,0 +1,60 @@
+The `logging` property configures the logging of the Topic Operator.
+For more details, see xref:con-entity-operator-logging-{context}[].
+
+
```
Do we need two empty lines?

scholzj commented on documentation/modules/con-kafka-logging.adoc:
```diff
@@ -0,0 +1,89 @@
+[id='con-kafka-logging-{context}']
+= Kafka loggers
+
+Kafka has its own configurable loggers:
```
I think we need to add here the Zookeeper logger as well: see the [old docs](https://strimzi.io/docs/latest/full.html#ref-loggers-Kafka-deployment-configuration-kafka)

ppatierno commented on documentation/modules/con-entity-operator-logging.adoc:
```diff
@@ -0,0 +1,79 @@
+Use the `logging` property to configure loggers and logger levels.
+
+You can set the log levels by specifying the logger and level directly (inline) or use a custom (external) ConfigMap.
+If a ConfigMap is used, it must be placed under under the `log4j2.properties` key.
```
typo, double "under"

ppatierno replied:
> If a ConfigMap is used, it must be placed under under the `log4j2.properties` key.

Tbh this statement could be misleading. It sounds to me that if you have a ConfigMap with the logging configuration, you have to put it (its name?) under the `log4j2.properties` key of the operator logging configuration.
Actually it is more like ...
In the `logging.name` property you have to set the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, you have to put the logging configuration under the `log4j2.properties`.

PaulRMellor replied:
Thanks Paolo. Much better. I will update each as: 

If a ConfigMap is used, you set `logging.name` property to the name of the ConfigMap containing the external logging configuration. Inside the ConfigMap, the logging configuration is described using `log4j2.properties`.

ppatierno commented on documentation/modules/con-kafka-bridge-logging.adoc:
```diff
@@ -0,0 +1,74 @@
+Use the `logging` property to configure loggers and logger levels.
+
+You can set the log levels by specifying the logger and level directly (inline) or use a custom (external) ConfigMap.
+If a ConfigMap is used, it must be placed under under the `log4j.properties` key.
```
ditto as above

PaulRMellor replied:
Fixed as above

ppatierno commented on documentation/modules/con-kafka-bridge-logging.adoc:
```diff
@@ -0,0 +1,74 @@
+  logging:
+    type: inline
+    loggers:
+      log4j.logger.io.strimzi.kafka.bridge: "INFO"
```
this is an example about the "root" logger but the description just mentions the OpenAPI operations loggers and not this one. You should add even this one to the above overview.

PaulRMellor replied:
I added it to the list and changed the introduction.

ppatierno commented on documentation/modules/con-kafka-connect-logging.adoc:
```diff
@@ -0,0 +1,54 @@
+Use the `logging` property to configure loggers and logger levels.
+
+You can set the log levels by specifying the logger and level directly (inline) or use a custom (external) ConfigMap.
+If a ConfigMap is used, it must be placed under under the `log4j.properties` key.
```
ditto as above

PaulRMellor replied:
Fixed as above

ppatierno commented on documentation/modules/con-kafka-connect-s2i-logging.adoc:
```diff
@@ -0,0 +1,54 @@
+Use the `logging` property to configure loggers and logger levels.
+
+You can set the log levels by specifying the logger and level directly (inline) or use a custom (external) ConfigMap.
+If a ConfigMap is used, it must be placed under under the `log4j.properties` key.
```
ditto as above

PaulRMellor replied:
Fixed as above

ppatierno commented on documentation/modules/con-kafka-logging.adoc:
```diff
@@ -0,0 +1,89 @@
+Use the `logging` property to configure loggers and logger levels.
+
+You can set the log levels by specifying the logger and level directly (inline) or use a custom (external) ConfigMap.
+If a ConfigMap is used, it must be placed under under the `log4j.properties` key according.
```
ditto as above

laidan6000 replied:
Repeated word: "under".

laidan6000 replied:
Accordingly? Or perhaps remove the word.

PaulRMellor replied:
Fixed as above

ppatierno commented on documentation/modules/con-kafka-logging.adoc:
```diff
@@ -0,0 +1,89 @@
+  # ...
+----
+
+Operators use the Apache `log4j2` logger implementation, so the ConfigMap must be placed under `log4j2.properties`.
```
One more time even here, it's not clear where is `log4j2.properties` key. It's inside the ConfigMap or it's inside the Kafka resource in the `logging` section. It's not the latter because it's also missing in the example and because ... it's just not :-)

PaulRMellor replied:
Yep. I've rewritten so we say a similar thing to the previous fix.

Operators use the Apache `log4j2` logger implementation, so the logging configuration is described inside the ConfigMap using `log4j2.properties`.

laidan6000 commented on documentation/modules/con-entity-operator-logging.adoc:
```diff
@@ -0,0 +1,79 @@
+
+The operators use the Apache `log4j2` logger implementation.
+
+Use the `logging` property to configure loggers and logger levels.
```
Suggested change:

Use the `logging` property in the `Kafka` resource to configure loggers and logger levels.

PaulRMellor replied:
Great idea! Done

laidan6000 commented on documentation/modules/con-entity-operator-logging.adoc:
```diff
@@ -0,0 +1,79 @@
+
+.Additional resources
+
+* Garbage collector (GC) logging can also be enabled (or disabled). For more information on GC, see xref:ref-jvm-options-deployment-configuration-kafka[]
```
Suggested change: For more information about GC logging, see xref:ref-jvm-options-deployment-configuration-kafka[].

PaulRMellor replied:
Great idea!

laidan6000 commented on documentation/modules/con-kafka-bridge-logging.adoc:
```diff
@@ -0,0 +1,74 @@
+z// This assembly is included in the following assemblies:
```
Replace the "z" with a forward slash.

laidan6000 commented on documentation/modules/con-kafka-bridge-logging.adoc:
```diff
@@ -0,0 +1,74 @@
+[id='con-kafka-bridge-logging-{context}']
+= Kafka Bridge loggers
+
+Kafka Bridge has a configurable logger for each OpenAPI operation, which are formatted as follows:
```
Suggested change:

Kafka Bridge has configurable loggers for each OpenAPI operation, which are formatted as follows:

PaulRMellor replied:
This section has changed slightly, as we now nclude another logger is the list.
So I changed to:

Kafka Bridge has its own configurable loggers:

* `log4j.logger.io.strimzi.kafka.bridge`
* `log4j.logger.http.openapi.operation.__<operation-id>__`

The `log4j.logger.http.openapi.operation.__<operation-id>__` logger is for each OpenAPI operation, where `_<operation-id>_` is the identifier of a specific operation.

Here are the operations defined by the OpenAPI specification:

laidan6000 commented on documentation/modules/con-kafka-bridge-logging.adoc:
```diff
@@ -0,0 +1,74 @@
+
+.Additional resources
+
+* Garbage collector (GC) logging can also be enabled (or disabled). For more information on GC, see xref:ref-jvm-options-deployment-configuration-kafka[]
```
Suggested change:

For more information about GC logging, see xref:ref-jvm-options-deployment-configuration-kafka[].

PaulRMellor replied:
Fixed

laidan6000 commented on documentation/modules/con-kafka-connect-logging.adoc:
```diff
@@ -0,0 +1,54 @@
+
+.Additional resources
+
+* Garbage collector (GC) logging can also be enabled (or disabled). For more information on GC, see xref:ref-jvm-options-deployment-configuration-kafka[]
```
Same as my previous suggestion.

PaulRMellor replied:
Fixed

laidan6000 commented on documentation/modules/con-kafka-connect-s2i-logging.adoc:
```diff
@@ -0,0 +1,54 @@
+
+.Additional resources
+
+* Garbage collector (GC) logging can also be enabled (or disabled). For more information on GC, see xref:ref-jvm-options-deployment-configuration-kafka[]
```
Same as previous suggestion.

PaulRMellor replied:
Fixed

laidan6000 commented on documentation/modules/con-kafka-logging.adoc:
```diff
@@ -0,0 +1,89 @@
+
+.Additional resources
+
+* Garbage collector (GC) logging can also be enabled (or disabled). For more information on garbage collection, see xref:ref-jvm-options-deployment-configuration-kafka[]
```
Same as previous suggestion.

PaulRMellor replied:
Fixed

laidan6000 commented on documentation/using/master.adoc:
```diff
@@ -35,7 +35,6 @@ include::modules/proc-uninstalling.adoc[leveloffset=+1]
 
 include::modules/proc-accessing-resource-status.adoc[leveloffset=+1]
 
-include::modules/ref-kafka-logging.adoc[leveloffset=+1]
```
Just checking that there is nothing to add to the `master.adoc` file.

PaulRMellor replied:
No. We just removed the appendix.

PaulRMellor commented:
@scholzj @ppatierno - updated following review

PaulRMellor commented:
@ppatierno - as discussed, I have merged the two logging sections in commit #4 and removed the logging section that was in the Kafka Bridge overview. I also removed the configmap snips that are no longer needed.

scholzj commented on documentation/modules/con-kafka-connect-logging.adoc:
```diff
@@ -0,0 +1,54 @@
+Kafka Connect has its own configurable loggers:
+
+* `connect.root.logger.level`
+* `log4j.logger.org.apache.zookeeper`
```
Connect doesn't connect to Zookeeper. Are you sure it has `log4j.logger.org.apache.zookeeper` and `log4j.logger.org.I0Itec.zkclient` loggers? Or is this just some bug copied from previous version of the doc?

PaulRMellor replied:
Certainly copied from the previous doc. I see these loggers described in relation to Kafka Connect: https://fossies.org/linux/kafka/config/connect-log4j.properties

scholzj replied:
@tombentley I get it that reflections logger might be used. But do you have any idea why would Conenct use Zookeeper and zkClient loggers? I think this must be a mistake in the Kafka files, or?

tombentley replied:
After a bit of digging... the zookeeper lines in the connect log4j seem to date back to 2015. It seems likely to me that at that time Connect did connect to ZK, but it doesn't do so any more. So I think we should remove this from the example.

PaulRMellor replied:
@scholzj @tombentley - Thanks. I have removed `log4j.logger.org.apache.zookeeper` and `log4j.logger.org.I0Itec.zkclient` loggers from the lists of Kafka Connect loggers and Kafka Connect with S2I loggers

scholzj commented on documentation/modules/con-kafka-connect-s2i-logging.adoc:
```diff
@@ -0,0 +1,54 @@
+Kafka Connect with Source2Image support has its own configurable loggers:
+
+* `connect.root.logger.level`
+* `log4j.logger.org.apache.zookeeper`
```
Same as above for the non-S2I implementation.

PaulRMellor replied:
Removed

scholzj commented on documentation/modules/con-kafka-logging.adoc:
```diff
@@ -0,0 +1,90 @@
+* `log4j.logger.kafka.log.LogCleaner`
+* `log4j.logger.state.change.logger`
+* `log4j.logger.kafka.authorizer.logger`
+* `zookeeper.root.logger`
```
Is this an actual logger for KAfka? Or is this only for Zookeeper? I guess the other. I think we need to split this into two separate lists - for KAfka and for Zoo.

PaulRMellor replied:
I split it off from the list.
